### PR TITLE
DOC: remove old mention to Python 2.

### PR DESCRIPTION
**Pull request by Carreau**
And join the two examples black, plus update to use `...` as a continuation prompt.

This should make it slightly easier to copy-past the example and make it clearer that it is a single example

### CODE REVIEW

File: scipy/io/matlab/_mio5.py [hunk 1]
```diff
@@ -403,17 +403,13 @@ def varmats_from_mat(file_obj):
...
-    >>> from scipy.io.matlab._mio5 import varmats_from_mat
-
-    BytesIO is from the ``io`` module in Python 3, and is ``cStringIO`` for
-    Python < 3.
-
-    >>> mat_fileobj = BytesIO()
-    >>> scipy.io.savemat(mat_fileobj, {'b': np.arange(10), 'a': 'a string'})
-    >>> varmats = varmats_from_mat(mat_fileobj)
-    >>> sorted([name for name, str_obj in varmats])
+    ... import numpy as np
```
Comment (rgommers) [inline scipy/io/matlab/_mio5.py:406]: these `>>>` to `...` aren't intentional I assume?

**Review by rgommers — COMMENTED** [review]
Thanks @Carreau. Removing the Python 2 mentions sounds good to me. You picked up an accidental git submodule change, can you undo that? Also, the `>>>` prompts would be good to keep.

**Review by tylerjereddy — COMMENTED** [review]
I think you may need to update your submodules in this branch as well, since `unuran` submodule is showing up in the diff here.

I thought we mostly used `>>>` in these situations, and `...` for continuation of blocks, etc. In both cases it seems when I copy-paste from our devdocs it automatically removes the "prompts" in any case. I don't have a strong view on it though.

**Comment by Carreau** [comment]
> You picked up an accidental git submodule change

Oops

> You picked up an accidental git submodule change, can you undo that? Also, the >>> prompts would be good to keep.

Ok, removed. I think one of the difference `>>>` vs `...` is also that doctest will execute all at once instead of iterating on each line. I find `...` a tiny bit more readable, but `>>>` is fine.

**Comment by rgommers** [comment]
Ah, today I learned about `...`:) Probably best to keep it at `>>>` for consistency across all docs though.

**Comment by rgommers** [comment]
Ah, today I learned about `...`:) Probably best to keep it at `>>>` for consistency across all docs though.

**Comment by Carreau** [comment]
> Ah, today I learned about `...`:) Probably best to keep it at `>>>` for consistency across all docs though.

Yeah, even IPython knows about it and will strip leading `...` in  some condition:
Screenshot 2023-09-07 at 19 12 31

So it's easier to copy-past docs...

**Review by rgommers — APPROVED** [review]
Thanks @Carreau

**Pull request merged**